### ISSUE TITLE
Files in /tmp/minimega/files on deployed nodes are not updating

### ISSUE THREAD

**Issue by derhart** [sandia-minimega/minimega#526]
We deploy an experiment, see an issue with some of our injected config files, then tear down and redeploy the system again. The files in /tmp/minimega/files, specifically snapshot images, do not update when there is a change to the contents of the snapshot image file. As discussed with @djfritz, this is because of the simple filename comparison that is currently used. It'd be great if there was a "dirty" flag that would inform the remote node that there is a new copy of the file in question. It would then stay in sync with the latest changes.

**Comment by jcrussell** [comment]
Perhaps we could add a `file sync <dir>` API that walks the directory and deletes outdated files based on the last modified time? We could make this interactive (with a `force` option).

**Comment by djfritz** [comment]
The right approach here is to use MSI, MESI, MOSI... coherence protocols that are designed for exactly these scenarios. It's a bigger lift than it may seem at first because of write-after-write and read-after-write races caused by multiple nodes having copies of the file. Short of having a distributed lock on /file access/ we're going to have to draw some lines around how far we want a coherency protocol to go.

**Comment by djfritz** [comment]
That said we should have something in place.

**Comment by derhart** [comment]
Do you have an idea when this feature might be implemented?

**Comment by djfritz** [comment]
No, though it could be a good intern project this summer. The iomeshage package is fairly compact and easy to digest. I'll poll for interest.

**Comment by djfritz** [comment]
There's no way this meets a 2.6 goal. Punting.

**Comment by GhostofGoes** [comment]
Not sure if this is still an issue 10 years later, leaving open to let AI investigate at a later date
